### PR TITLE
[feat]: [PL-31460]: PreQA hotifx bugfix for NG enablement

### PR DESCRIPTION
**Pull request by shashank-harness**
## Describe your changes
- When we enable NG for an account, migration of CG users to NG starts. We have handled that scenario for new userAccountLevelData generation.
- Another case is when a user gets added in NG then we need to update the userAccountLevelData on CG side. We have fixed a missing case in that.

**Testing:**
- All scenarios tested on Local.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46501)

### CODE REVIEW

**Comment by shashank-harness** [comment]
trigger ut2

**Comment by shashank-harness** [comment]
trigger ut2

**Review by tiwaritejas — APPROVED** [review]
> Reviewed 7 of 9 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46501)* status: 10 file(s) to review. 1 of 2 approvals obtained.

File: 120-ng-manager/src/main/java/io/harness/ng/core/event/NGAccountSetupService.java [hunk 1]
```diff
@@ -308,6 +309,7 @@ private void upsertUserMembership(Scope scope, String userId) {
               .projectIdentifier(scope.getProjectIdentifier())
               .build(),
           emptyList(), emptyList(), UserMembershipUpdateSource.SYSTEM);
+      ngUserService.updateNGUserToCGWithSource(userId, scope, MANUAL);
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/event/NGAccountSetupService.java:312]: Can we add testing done for this change. This is changing existing behaviour.

**Review by meenakshi-harness — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/event/NGAccountSetupService.java (hunk 1 above)
Comment (shashank-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/event/NGAccountSetupService.java:312]: I tested by creating a MANUAL user in NG and updating it via SCIM and it updated successfully

**Pull request merged**